RecoMuon/MuonIdentification/plugins/cuts/MuonTrackCut.cc

```diff
@@ -0,0 +1,116 @@
+    if ( cc.exists("minNumberOfValidPixelHits") ) minNumberOfValidPixelHits_ = cc.getParameter<int>("minNumberOfValidPixelHits");
+    if ( cc.exists("minValidFraction") ) minValidFraction_ = cc.getParameter<double>("minValidFraction");
+    const std::string trackQualityStr = cc.exists("trackQuality") ? cc.getParameter<std::string>("trackQuality") : "";
+    trackQuality_ = reco::Track::qualityByName(trackQualityStr);
+  }
+  if ( c.existsAs<edm::ParameterSet>("globalTrack") )
+  {
+    doGlobalTrack_ = true;
+
+    const edm::ParameterSet cc = c.getParameter<edm::ParameterSet>("globalTrack");
+    //if ( cc.exists("maxNormalizedChi2") ) maxNormalizedChi2_ = cc.getParameter<double>("maxNormalizedChi2");
+    if ( cc.exists("minNumberOfValidMuonHits") ) minNumberOfValidMuonHits_ = cc.getParameter<int>("minNumberOfValidMuonHits");
+  }
+
+}
+
+// Functors for evaluation
+CutApplicatorBase::result_type MuonTrackCut::operator()(const reco::MuonPtr& muon) const
+{
+  if ( doInnerTrack_ )
+  {
+    const reco::TrackRef t = muon->innerTrack();
+    if ( t.isNull() ) return false;
+    const auto& h = t->hitPattern();
+    if ( trackQuality_ != reco::Track::undefQuality and !t->quality(trackQuality_) ) return false;
+    if ( h.trackerLayersWithMeasurement() < minTrackerLayersWithMeasurement_ ) return false;
+    if ( h.pixelLayersWithMeasurement() < minPixelLayersWithMeasurement_ ) return false;
+    if ( h.numberOfValidPixelHits() < minNumberOfValidPixelHits_ ) return false;
+    if ( t->validFraction() <= minValidFraction_ ) return false;
+  }
+  if ( doGlobalTrack_ )
+  {
+    const reco::TrackRef t = muon->globalTrack();
+    if ( t.isNull() ) return false;
+    const auto& h = t->hitPattern();
+    if ( h.numberOfValidMuonHits() < minNumberOfValidMuonHits_ ) return false;
+    // if ( t->normalizedChi2() > maxNormalizedChi2_ ) return false; Not used for 
+  }
+
+  return true;
+}
+
+double MuonTrackCut::value(const reco::CandidatePtr& cand) const
+{
+  const reco::MuonPtr muon(cand);
+  if ( doInnerTrack_ )
+  {
+    const reco::TrackRef t = muon->innerTrack();
+    if ( t.isNull() ) return false;
+    const auto& h = t->hitPattern();
+    if ( trackQuality_ != reco::Track::undefQuality and !t->quality(trackQuality_) ) return t->quality(trackQuality_);
+    if ( h.trackerLayersWithMeasurement() < minTrackerLayersWithMeasurement_ ) return h.trackerLayersWithMeasurement();
+    if ( h.pixelLayersWithMeasurement() < minPixelLayersWithMeasurement_ ) return h.pixelLayersWithMeasurement();
+    if ( h.numberOfValidPixelHits() < minNumberOfValidPixelHits_ ) return h.numberOfValidPixelHits();
+    if ( t->validFraction() <= minValidFraction_ ) return t->validFraction();
+
+    return t->validFraction();
+  }
+  if ( doGlobalTrack_ )
+  {
+    const reco::TrackRef t = muon->globalTrack();
+    if ( t.isNull() ) return false;
```

Comment:
This method has a return type of "double". It is confusing to the reader for the method to return "false", which is a "bool". Please change to "return 0;" here and on line 94.